.github/workflows/Actions-workflow.yml

```diff
@@ -0,0 +1,241 @@
+            markdownFile.write('# GitHub Actions Workflow Overview\n\n')
+            sortedDictionary=dict(sorted(workflowDetails.items(),key=lambda item:item[1]['name']))
+            for file in sortedDictionary:
+              uniqFileName=workflowDetails[file]['name']
+              uniqFileName=uniqFileName.replace(' ','_')      #replace spaces with underscore
+              uniqFileName=uniqFileName.replace('(','_')      #replace parenthesis with underscore
+              uniqFileName=uniqFileName.replace(')','_')
+              uniqFileName=uniqFileName.upper()
+              fileBlock=f"{uniqFileName}[\"{workflowDetails[file]['name']}[{file}]\"]"
+              jobs=workflowDetails[file]['jobs']
+              markdownFile.write('```mermaid\n')
+              markdownFile.write('\nflowchart LR\n')
+              for job in jobs:
+                needs=jobs[job]['needs']
+                needs=needs.replace('[','')
+                needs=needs.replace(']','')
+                needs=needs.replace(' ','')
+                if len(needs) == 0:
+                  markdownFile.write(f"\n  {fileBlock} ---> {job}({job})")
+                else:
+                  needs=list(needs.split(','))
+                  for need in needs:
+                    markdownFile.write(f"\n  {need}({need}) ---> {job}({job})")
+                uses=jobs[job]['uses']
+                if uses!="":
+                  uses=uses.split('/')[-1]
+                  str=","
+                  inputs=str.join(jobs[job]['inputs'])
+                  inputs=inputs.replace('\"','')    #remove " (double quotes)
+                  inputs=inputs.replace("\'",'')    #remove ' (single quotes)
+                  if len(inputs)!=0:
+                    markdownFile.write(f"\n  {job}(\"{job}\ninputs:[{inputs}]\") ---> {uses}(\"{workflowDetails[uses]['name']}[{uses}]\")")
+                  else:
+                    markdownFile.write(f"\n  {job}({job}) ---> {uses}")
+              
+              markdownFile.write('\n\n```\n\n')
+        
+          with open( markdown ,'a') as markdownFile:
+            markdownFile.write('## GitHub Actions Trigger Overview\n')
+            triggersList=set()
+            for file in workflowDetails:
+              for trigger in workflowDetails[file]['trigger']:
+                triggersList.add(trigger)  #triggersList contains unique list of all types of triggers used in the workflow file
+                    
+            for trigger in sorted(triggersList):
+              markdownFile.write('\n```mermaid\n')
+              markdownFile.write('\nflowchart TB\n')
+              markdownFile.write(f'\n  subgraph {trigger.upper()}')
+              for file in sortedDictionary:
+                if trigger in workflowDetails[file]['trigger']:
+                  markdownFile.write(f'\n    {trigger} ---> {file}[\"{workflowDetails[file]["name"]}[{file}]\"]')
+              markdownFile.write('\n  end\n')
+              markdownFile.write('\n```\n')
```

Comment:
The close is missing for the markdown file. It is a good practice if you open a file then close when not used anymore.